themes/gatsby-theme-blog/gatsby-config.js

```diff
@@ -1,88 +1,65 @@
+        github: `https://github.com/gatsbyjs/gatsby`,
       },
     },
-    {
-      resolve: `gatsby-source-filesystem`,
-      options: {
-        name: `blog-default-assets`,
-        path: `${__dirname}/content/assets`,
-      },
-    },
-    `gatsby-transformer-sharp`,
-    `gatsby-plugin-sharp`,
-    {
-      resolve: `gatsby-plugin-mdx`,
-      options: {
-        extensions: [`.mdx`, `.md`],
-        gatsbyRemarkPlugins: [
-          {
-            resolve: `gatsby-remark-images`,
-            options: {
-              // should this be configurable by the end-user?
-              maxWidth: 1380,
-              linkImagesToOriginal: false,
+    plugins: [
+      mdx && {
+        resolve: `gatsby-plugin-mdx`,
+        options: {
+          extensions: [`.mdx`, `.md`],
+          gatsbyRemarkPlugins: [
+            {
+              resolve: `gatsby-remark-images`,
+              options: {
+                // should this be configurable by the end-user?
+                maxWidth: 1380,
+                linkImagesToOriginal: false,
+              },
             },
-          },
-          { resolve: `gatsby-remark-responsive-iframe` },
-          { resolve: `gatsby-remark-copy-linked-files` },
-          { resolve: `gatsby-remark-numbered-footnotes` },
-          { resolve: `gatsby-remark-smartypants` },
-          { resolve: `gatsby-remark-code-titles` },
-          {
-            resolve: `gatsby-remark-prismjs`,
-            options: {
-              noInlineHighlight: true,
+            { resolve: `gatsby-remark-responsive-iframe` },
+            { resolve: `gatsby-remark-copy-linked-files` },
+            { resolve: `gatsby-remark-numbered-footnotes` },
+            { resolve: `gatsby-remark-smartypants` },
+            { resolve: `gatsby-remark-code-titles` },
+            {
+              resolve: `gatsby-remark-prismjs`,
+              options: {
+                noInlineHighlight: true,
+              },
             },
-          },
-        ],
-        remarkPlugins: [require(`remark-slug`)],
+          ],
+          remarkPlugins: [require(`remark-slug`)],
+        },
       },
-    },
-    `gatsby-plugin-react-helmet`,
-    `gatsby-plugin-twitter`,
-    {
-      resolve: `gatsby-plugin-page-creator`,
-      options: {
-        path: `${__dirname}/src/pages`,
+      {
+        resolve: `gatsby-source-filesystem`,
+        options: {
+          path: options.contentPath || `content/posts`,
+          name: options.contentPath || `content/posts`,
+        },
       },
-    },
-    `gatsby-plugin-emotion`,
-  ],
+      {
+        resolve: `gatsby-source-filesystem`,
+        options: {
+          path: options.assetPath || `content/assets`,
+          name: options.assetPath || `content/assets`,
+        },
+      },
+      `gatsby-transformer-sharp`,
+      `gatsby-plugin-sharp`,
+      `gatsby-plugin-react-helmet`,
+      `gatsby-plugin-twitter`,
+      `gatsby-plugin-emotion`,
+    ],
```

Comment:
I _think_ we're going to want to add a filter here in case `mdx` is set to false.

```suggestion
    ].filter(Boolean),
```